src/main/java/io/jenkins/plugins/mcp/server/extensions/BuildArtifactsExtension.java

```diff
@@ -0,0 +1,192 @@
+import io.jenkins.plugins.mcp.server.annotation.Tool;
+import io.jenkins.plugins.mcp.server.annotation.ToolParam;
+import jakarta.annotation.Nullable;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import jenkins.util.VirtualFile;
+import lombok.extern.slf4j.Slf4j;
+
+@Extension
+@Slf4j
+public class BuildArtifactsExtension implements McpServerExtension {
+
+    /*
+     * Suppression Rationale:
+     * - "rawtypes": Jenkins' Run.Artifact is a raw type in Jenkins' legacy API
+     * - "unchecked": Run.getArtifacts() returns raw List requiring unchecked conversion
+     */
+    @SuppressWarnings({"rawtypes", "unchecked"})
+    @Tool(description = "Get the artifacts for a specific build or the last build of a Jenkins job")
+    public List<Run.Artifact> getBuildArtifacts(
+            @ToolParam(description = "Job full name of the Jenkins job (e.g., 'folder/job-name')") String jobFullName,
+            @Nullable
+                    @ToolParam(
+                            description = "Build number (optional, if not provided, returns the last build)",
+                            required = false)
+                    Integer buildNumber) {
+        return getBuildByNumberOrLast(jobFullName, buildNumber)
+                .map(Run::getArtifacts)
+                .orElse(List.of());
+    }
+
+    @Tool(
+            description =
+                    "Get the content of a specific build artifact with pagination support. Returns the artifact content as text with information about whether there is more content to retrieve.")
+    public BuildArtifactResponse getBuildArtifact(
+            @ToolParam(description = "Job full name of the Jenkins job (e.g., 'folder/job-name')") String jobFullName,
+            @Nullable
+                    @ToolParam(
+                            description = "Build number (optional, if not provided, returns the last build)",
+                            required = false)
+                    Integer buildNumber,
+            @ToolParam(description = "Relative path of the artifact from the artifacts root") String artifactPath,
+            @Nullable
+                    @ToolParam(
+                            description =
+                                    "The byte offset to start reading from (optional, if not provided, starts from the beginning)",
+                            required = false)
+                    Long offset,
+            @Nullable
+                    @ToolParam(
+                            description =
+                                    "The maximum number of bytes to read (optional, if not provided, reads up to 64KB)",
+                            required = false)
+                    Integer limit) {
+
+        if (offset == null || offset < 0) {
+            offset = 0L;
+        }
+        if (limit == null || limit <= 0) {
+            limit = 65536; // 64KB default
```

Comment:
might be better having a configuration (e.g SystemProperties) as users may want to configure more or less.